baseapp/baseapp.go

```diff
@@ -383,6 +381,8 @@ func (app *BaseApp) BeginBlock(req abci.RequestBeginBlock) (res abci.ResponseBeg
 	// if this is a test and InitChain was never called.
 	if app.deliverState == nil {
 		app.setDeliverState(req.Header)
+	} else {
+		app.deliverState.ctx = app.deliverState.ctx.WithBlockHeader(req.Header)
```

Comment:
Why is this now necessary? `app.DeliverState` is set to nil when `Commit()` is called. Also, I think the comment on line 381 is wrong for this reason.